### PR TITLE
Introduce ServiceResourceHelperService - small service that prepares …

### PR DESCRIPTION
**Pull request by RomanTeren**
…response for ServiceResourceApiImpl

Add @LogicExtensionPoint to allow to customize response, for example, add custom service characteristics in leps.

### CODE REVIEW

**Review by icontributor-xm — APPROVED** [review]

**Review by pkev777 — APPROVED** [review]

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import com.icthh.xm.tmf.ms.activation.model.v4.Service;
+import com.icthh.xm.tmf.ms.activation.model.v4.ServiceCreate;
+import com.icthh.xm.tmf.ms.activation.resolver.TransactionTypeKeyResolver;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+@LepService(group = "service.resource")
+@org.springframework.stereotype.Service
```
Comment (MaximBogun) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperServiceImpl.java:16]: Pls use import.

Comment (RomanTeren) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperServiceImpl.java:16]: already used for Service below

File: src/test/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImplTest.java [hunk 2]
```diff
@@ -39,7 +40,7 @@
 @RunWith(SpringRunner.class)
 @WebMvcTest(controllers = ServiceResourceApiController.class, secure = false)
 @ContextConfiguration(classes = {ServiceResourceApiController.class, ServiceResourceApiImpl.class,
-    ExceptionTranslator.class, ServiceMapperImpl.class})
+    ExceptionTranslator.class, ServiceResourceHelperServiceImpl.class, ServiceMapperImpl.class})
 public class ServiceResourceApiImplTest {
```
Comment (VictorKhoroshko) [inline src/test/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImplTest.java:44]: Can we extend tests with new functionality?

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+public class ServiceResourceHelperServiceImpl implements ServiceResourceHelperService {
+
+    private final ServiceMapper serviceMapper;
+
+    @Override
+    @LogicExtensionPoint(value = "EnrichServiceResponse", resolver = TransactionTypeKeyResolver.class)
+    public Service enrichServiceResponse(ServiceCreate serviceCreate, SagaTransaction sagaTransaction) {
+        Service service = serviceMapper.serviceCreateToService(serviceCreate);
```
Comment (MaximBogun) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperServiceImpl.java:24]: You have Slf4j, but you don't use log

**Review by MaximBogun — APPROVED** [review]

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperService.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.icthh.xm.tmf.ms.activation.service;
+
+import com.icthh.xm.tmf.ms.activation.domain.SagaTransaction;
+import com.icthh.xm.tmf.ms.activation.model.v4.Service;
+import com.icthh.xm.tmf.ms.activation.model.v4.ServiceCreate;
+
+public interface ServiceResourceHelperService {
```
Comment (VictorKhoroshko) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperService.java]: Can we invent some other meaningful name without _Helper_ and _Service*Service_ like _ResponseEnricher_ or smth?

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperServiceImpl.java (hunk 3 above)
Comment (RomanTeren) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperServiceImpl.java:22]: removed the annotation

**Review by VictorKhoroshko — APPROVED** [review]

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperService.java (hunk 4 above)
Comment (RomanTeren) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/ServiceResourceHelperService.java]: done

File: src/test/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImplTest.java (hunk 2 above)
Comment (RomanTeren) [reply src/test/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImplTest.java:44]: I've got 100% coverage for all the changes: 
image

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/ResponseEnricherServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import com.icthh.xm.tmf.ms.activation.domain.SagaTransaction;
+import com.icthh.xm.tmf.ms.activation.mapper.ServiceMapper;
+import com.icthh.xm.tmf.ms.activation.model.v4.Service;
+import com.icthh.xm.tmf.ms.activation.model.v4.ServiceCreate;
+import com.icthh.xm.tmf.ms.activation.resolver.TransactionTypeKeyResolver;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@LepService(group = "service.resource")
+@org.springframework.stereotype.Service
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/tmf/ms/activation/service/ResponseEnricherServiceImpl.java:14]: @RomanTeren 
`@LepService` already contains spring `@Service`. don not need to add again

File: src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImpl.java [hunk 6]
```diff
@@ -30,10 +30,9 @@
     private static final String RELATED_PARTY_ID = "relatedParty.id";
     private static final String RELATED_PARTY_REFERRED_TYPE = "relatedParty.referredType";
 
-
-    private final ServiceMapper serviceMapper;
     private final SagaService sagaService;
     private final SagaTransactionFactory sagaTransactionFactory;
+    private final ResponseEnricherServiceImpl responseEnricher;
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImpl.java:35]: @RomanTeren why don't you use the interface here?

Comment (RomanTeren) [reply src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImpl.java:35]: my bad, I'll fix it separate PR

File: src/main/java/com/icthh/xm/tmf/ms/activation/service/ResponseEnricherServiceImpl.java (hunk 5 above)
Comment (RomanTeren) [reply src/main/java/com/icthh/xm/tmf/ms/activation/service/ResponseEnricherServiceImpl.java:14]: https://github.com/xm-online/tmf-ms-activation/pull/94

File: src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImpl.java (hunk 6 above)
Comment (RomanTeren) [reply src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/v4/ServiceResourceApiImpl.java:35]: https://github.com/xm-online/tmf-ms-activation/pull/94

**Pull request merged**